client/lib/olark/index.js

```diff
@@ -368,8 +372,28 @@ const olark = {
 			this.userType = userType;
 			return true;
 		} );
+	},
+
+	notifyLocation( location ) {
+		const { details } = olarkStore.get();
+
+		if ( ! details.isConversing ) {
+			return;
+		}
+
+		olarkActions.sendNotificationToOperator( 'looking at ' + location );
 	}
 };
 
 emitter( olark );
 olark.initialize();
+
+export function notifyLocation( context, next ) {
+	// Use context.path here because window.location.href has the previous location and not what the new location will be.
+	const location = `${window.location.protocol}//${window.location.host}${context.path}`;
```

Comment:
In case we could run into issues with other sneaky parts of the URL, I'm pretty sure we could use the `url` library here and only replace the part we want instead of rebuilding from scratch…

```js
import url from 'url';

const location = url.resolve( url.parse( window.location ), context.canonicalPath );
```

This seems to work in my node console and then we wouldn't have to worry about protocol or port or query string or anything like that.